tests/GrapheneOS/ConnectivityCheckSettingTest/src/grapheneos/test/ConnectivityCheckSettingTest.java

```diff
@@ -0,0 +1,147 @@
+        public final String httpsUrl;
+
+        ConnectivityCheckServer(String httpUrl, String httpsUrl) {
+            this.httpUrl = httpUrl;
+            this.httpsUrl = httpsUrl;
+        }
+    }
+
+    private void toggleWifi(String state) throws DeviceNotAvailableException {
+        String cmd = String.format(WIFI_CMD, state);
+        getDevice().executeShellV2Command(cmd);
+    }
+
+    @Before
+    public void setUp() throws DeviceNotAvailableException {
+        if (!getDevice().isAdbRoot()) {
+            try {
+                getDevice().enableAdbRoot();
+            } finally {
+                fail("adb root access is required");
+            }
+        }
+        getDevice().clearLogcat();
+    }
+
+    private final ArrayList<String> parseLogcat(String regexp, int setting) throws Exception {
+        final Pattern pattern = Pattern.compile(regexp);
+        ArrayList<String> result = new ArrayList<String>();
+
+        try (InputStreamSource logSource = getDevice().getLogcat()) {
+            InputStreamReader streamReader = new InputStreamReader(logSource.createInputStream());
+            BufferedReader logReader = new BufferedReader(streamReader);
+
+            String line;
+            while ((line = logReader.readLine()) != null) {
+                Matcher matcher = pattern.matcher(line);
+                if (setting != DISABLED_SETTING) {
+                    if (matcher.find() && line.contains(LOG_TAG)) {
+                        result.add(line);
+                        if (result.toString().contains("PROBE_HTTP") && result.toString().contains("PROBE_HTTPS")) {
+                            break;
+                        }
+                    }
+                } else {
+                    if (matcher.find()) {
+                        result.add(matcher.group(0));
+                        break;
+                    }
+                }
+            }
+            return result;
+        }
+    }
+
+    private final boolean checkParsed(ArrayList<String> matched, int setting) {   
+        boolean result = switch (setting) {
+            case GRAPHENEOS_SERVER_SETTING ->
+                    matched.get(0).contains(ConnectivityCheckServer.GRAPHENEOS.httpUrl) && matched.get(1).contains(ConnectivityCheckServer.GRAPHENEOS.httpsUrl);
+            case STANDARD_SERVER_SETTING ->
+                    matched.get(0).contains(ConnectivityCheckServer.STANDARD.httpUrl) && matched.get(1).contains(ConnectivityCheckServer.STANDARD.httpsUrl);
+            case DISABLED_SETTING -> matched.get(0).equals(FILTER_DISABLED_REGEXP);
+            default -> false;
+        };
+        return result;
+    }
+
+    private void baseTest(int setting) throws Exception {
+        toggleWifi("disabled");
+        getDevice().setProperty(SYS_PROP, Integer.toString(setting));
+        RunUtil.getDefault().sleep(1000); // 1s
+        getDevice().clearLogcat();
+        toggleWifi("enabled");
+        RunUtil.getDefault().sleep(10000); // 10s
```

Comment:
This statement slows down the test massively, it's repeated for each of three cases. In most cases connectivity check completes much faster.

Better to use a loop:
```
boolean hasConnectivity = false;
for (int i = 0; i < 10; ++i) {
   sleep(1000)
   hasConnectivity = getDevice().checkConnectivity()
   if (hasConnectivity) {
      break
   }
}
assertTrue(hasConnectivity)
```